sdk/identity/azure-identity/azure/identity/_credentials/vscode_credential.py

```diff
@@ -43,11 +46,15 @@ def get_token(self, *scopes, **kwargs):
         if not scopes:
             raise ValueError("'get_token' requires at least one scope")
 
-        refresh_token = get_credentials()
-        if not refresh_token:
-            raise CredentialUnavailableError(
-                message="No Azure user is logged in to Visual Studio Code."
-            )
-        token = self._client.get_cached_access_token(scopes) \
-                or self._client.obtain_token_by_refresh_token(refresh_token, scopes, **kwargs)
+        token = self._client.get_cached_access_token(scopes)
+
+        if token:
+            return token
+
+        if not self._refresh_token:
+            self._refresh_token = get_credentials()
+        if not self._refresh_token:
+            raise CredentialUnavailableError(message="No Azure user is logged in to Visual Studio Code.")
```

Comment:
This checks the conditional twice when we do have a refresh token. We only need the second check after calling `get_credentials`.

Also, just a note for the sake of thoroughness: this still allows unexpected authentication in the case that a user signs in between two calls to `get_token`. I think that's okay so long as this credential is only exposed through `DefaultAzureCredential`, which won't try an unavailable credential again after succeeding with another.